lib/matplotlib/figure.py

```diff
@@ -1377,14 +1380,24 @@ def add_subplot(self, *args, **kwargs):
                 raise TypeError('Positional arguments are not a valid '
                                 'position specification.')
         elif nargs == 3:
-            for arg in args:
+            newarg = [None] * 3
+            message = ("Passing non-integers as three-element "
+                       "position specification is deprecated since "
+                       "%(since)s and will be removed %(removal)s.")
+            for nn, arg in enumerate(args[:2]):
                 if not isinstance(arg, Integral):
-                    cbook.warn_deprecated(
-                        "3.3",
-                        message="Passing non-integers as three-element "
-                                "position specification is deprecated since "
-                                "%(since)s and will be removed %(removal)s.")
-            args = tuple(map(int, args))
+                    cbook.warn_deprecated("3.3", message=message)
+                newarg[nn] = int(arg)
+            if isinstance(args[2], tuple):
+                # start/stop two-tuple is allowed...
+                for arg in args[2]:
+                    if not isinstance(arg, Integral):
+                        cbook.warn_deprecated("3.3", message=message)
+                newarg[nn] = (int(args[0]), int(args[1]))
```

Comment:
Needs a length check then, as it accepts `(1, 2, 3, 4)` and drops the last two.

Reply:
oops, yeah, and it was wrong too..

Reply:
eww, very wrong. Sorry about that.  I think its fixed now.  But checking this is a really ugly mess...  i.e. I'm not convinced the cure isn't worse than the disease.